File: final_back/app/Models/OrderItem.ts
import { DateTime } from 'luxon'
import { BaseModel, BelongsTo, belongsTo, column } from '@ioc:Adonis/Lucid/Orm'
import Order from './Order'
import Product from './Product'

export default class OrderItem extends BaseModel {
  public static table= "order_items"
  @column({ isPrimary: true })
  public id: number

  @column ({serializeAs: "order_id"})
  public orderId: number

  @column ({serializeAs: "product_id"})
  public productId: number

  @column ({serializeAs: "quantity"})
  public quantity: number

  @column ({serializeAs: "price"})
  public price: number

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime

  @belongsTo(()=>Order)
  public order: BelongsTo<typeof Order>
  
  @belongsTo (()=>Product)
  public product: BelongsTo<typeof Product>
}

File: final_project/lib/models/order_item.dart
class OrderItem {
  int id;
  int orderId;
  int productId;
  int quantity;
  int price;

  OrderItem(
    this.id,
    this.orderId,
    this.productId,
    this.quantity,
    this.price,
  );

  factory OrderItem.fromJson(json) {
    return OrderItem(
      json["id"] ?? 0,
      json["order_id"] ?? 0,
      json["product_id"] ?? 0,
      json["quantity"] ?? 0,
      int.parse(json["price"]) ?? 0,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "order_id": orderId,
      "product_id": productId,
      "quantity": quantity,
      "price": price,
    };
  }
}
